### ISSUE TITLE
Retire legacy code in torchtext

### ISSUE THREAD

**Issue by zhangguanheng66** [PyTorchLightning/pytorch-lightning#3529]
## ❓ Questions and Help

As the maintainer of the torchtext library, we plan to retire `torchtext.data.batch.Batch` and `torchtext.data.example.Example` in the next release (by the end of October). Those two components will be still available in `torchtext.legacy`. To handle the compatibility issue, should we send a PR and fix this on GitHub first, or we could just make some changes in fbcode?

Here are the proposed changes:
```
# old API
from torchtext.data.batch import Batch
from torchtext.data.example import Example

# new API
from torchtext.legacy.data.batch import Batch
from torchtext.legacy.data.example import Example
```

Relevant issue: https://github.com/pytorch/text/issues/985

**Comment by Borda** [comment]
I think when you know from which version you are going to change it we can implement it right away on one of the following ways (also if you have another proposal it is welcome)
1. with known version:
    ```
    if torchtext.__version__ < X.Y:
        from torchtext.data.batch import Batch
        from torchtext.data.example import Example
    else:
        from torchtext.legacy.data.batch import Batch
        from torchtext.legacy.data.example import Example
    ```
2. generic ways:
    ```
    try:
        from torchtext.legacy.data.batch import Batch
        from torchtext.legacy.data.example import Example
    except ImportError:
        from torchtext.data.batch import Batch
        from torchtext.data.example import Example
    ```

**Comment by zhangguanheng66** [comment]
> I think when you know from which version you are going to change it we can implement it right away on one of the following ways (also if you have another proposal it is welcome)
> 
> 1. with known version:
>    ```
>    if torchtext.__version__ < X.Y:
>        from torchtext.data.batch import Batch
>        from torchtext.data.example import Example
>    else:
>        from torchtext.legacy.data.batch import Batch
>        from torchtext.legacy.data.example import Example
>    ```
> 2. generic ways:
>    ```
>    try:
>        from torchtext.legacy.data.batch import Batch
>        from torchtext.legacy.data.example import Example
>    except ImportError:
>        from torchtext.data.batch import Batch
>        from torchtext.data.example import Example
>    ```

Thanks. We will know which the version we will change. I'm just wondering if I should change this on Github first, or I can send a Diff internally to fbcode and later on sync with the master branch. I just want to know the mechanism to sync PyTorchLightening between fbcode and github.

**Comment by Borda** [comment]
you can prepare a PR and later ping us when it actually, just not to forget it...
even throw we would find it as soon as you release it that some our tests are failing :]

**Comment by zhangguanheng66** [comment]
Sounds good. For the timeline, we will drop those legacy code from the master branch early October and eventually in 0.8.0 release (10/27/2020).

**Comment by Borda** [comment]
so I think that if you decided to go with version check you can send it now...

**Comment by ananthsub** [comment]
> Thanks. We will know which the version we will change. I'm just wondering if I should change this on Github first, or I can send a Diff internally to fbcode and later on sync with the master branch. I just want to know the mechanism to sync PyTorchLightening between fbcode and github.

@zhangguanheng66 There is none. PyTorch Lightning isn't linked to internal pytorch

**Comment by zhangguanheng66** [comment]
> > Thanks. We will know which the version we will change. I'm just wondering if I should change this on Github first, or I can send a Diff internally to fbcode and later on sync with the master branch. I just want to know the mechanism to sync PyTorchLightening between fbcode and github.
> 
> @zhangguanheng66 There is none. PyTorch Lightning isn't linked to internal pytorch

@ananthsub Thanks. So I have to manually fix the internal version by myself, right? Is there anyone importing OSS PyTorch Lightning to fbcode periodically?

**Comment by ananthsub** [comment]
That’s me + our team. We have scripts to do this

**Comment by Borda** [comment]
@zhangguanheng66 what is the progress here?

**Comment by zhangguanheng66** [comment]
> @zhangguanheng66 what is the progress here?

We are still cleaning up the new building blocks in torchtext so we postponed the retirement of the legacy code (maybe Nov. and December).

**Comment by Borda** [comment]
> > @zhangguanheng66 what is the progress here?
> 
> We are still cleaning up the new building blocks in torchtext so we postponed the retirement of the legacy code (maybe Nov. and December).

@zhangguanheng66 any update here, the freeze for `<0.7` becoming a blocker for us for an upgrade on python 3.9 as the first torch supporting  it is `torch>=1.7.1` and so your `torchtext==0.8.1`

**Comment by zhangguanheng66** [comment]
For now, you can use 0.8.1 for torchtext, although the legacy code is still there.

**Comment by zhangguanheng66** [comment]
We have landed the PR to retire the legacy code https://github.com/pytorch/text/pull/1172. This will be released in torchtext v0.9.0, and we are targeting the release date on March 4. 

If you still prefer to use `batch`, you can add the `legacy` tag to the path, a.k.a.
```
from torchtext.legacy.data.batch import Batch
```

**Comment by Borda** [comment]
we have some legacy separation in https://github.com/PyTorchLightning/pytorch-lightning/pull/6211/files
